.github/workflows/ibis-main.yml

```diff
@@ -76,6 +76,10 @@ jobs:
         if: matrix.os == 'windows-latest'
         run: choco install graphviz
 
+      - name: install minimum versions of dependencies
+        run: |
+          pip install -r ci/requirements/requirements-${{ matrix.python-version }}.txt
+
       - name: install ibis
         run: poetry install --without dev --without docs --extras "visualization decompiler"
```

Comment:
I'm getting `Package operations: 6 installs, 0 updates, 9 removals` so it looks like the lock file is overriding the minimum versions.

Reply:
Per https://github.com/python-poetry/poetry/issues/2861

> The easiest way to bypass poetry's locking, is to install the current project via pip: pip install . or pip install -e . for editable installs.

Reply:
An alternative: https://github.com/astral-sh/uv?tab=readme-ov-file#resolution-strategy